tt_eager/tt_dnn/op_library/softmax/softmax_op.hpp

```diff
@@ -47,6 +47,7 @@ struct Softmax {
     const bool inplace;
     const MemoryConfig output_mem_config;
     const tt::operations::primary::transformers::SoftmaxProgramConfig program_config;
+    const bool is_causal_mask;
```

Comment:
Do we need this attribute?

Can it be inferred by the op instead based on the mask shape?